cron/cron.go

```diff
@@ -0,0 +1,210 @@
+package cron
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/contribsys/faktory/client"
+	"github.com/contribsys/faktory/server"
+	"github.com/contribsys/faktory/util"
+	"github.com/robfig/cron/v3"
+)
+
+var _ server.Subsystem = &CronSubsystem{}
+
+// CronSubsystem enables cron jobs to queue faktory jobs
+// cron jobs are defined in the configuration
+type CronSubsystem struct {
+	statsDClient statsd.ClientInterface
+	Server       *server.Server
+	Options      *Options
+	Cron         *cron.Cron
+	EntryIDs     []cron.EntryID
+}
+
+type Options struct {
+	// Enabled controls whether or not the plugin will function
+	Enabled bool
+	// Tags sent to datdog
+	CronJobs []CronJob
+}
+
+type CronJob struct {
+	EntryId    cron.EntryID
+	Name       string
+	Schedule   string
+	faktoryJob []byte
+}
+
+// Start - parse config and schedule cron jobs
+func (c *CronSubsystem) Start(s *server.Server) error {
+	c.Server = s
+	options, err := c.getOptions(s)
+	// only throw an error if the config is invalid and the plugin is enabled
+	c.Options = options
+	if err != nil && c.Options.Enabled {
+		return fmt.Errorf("Start: Error parsing config: %v", err)
+	}
+
+	if !c.Options.Enabled {
+		return nil
+	}
+
+	c.createCron()
+
+	if err := c.addCronJobs(); err != nil {
+		return fmt.Errorf("Start: cannot add cron jobs: %v", err)
+	}
+
+	util.Infof("Started cron plugin, registered %d jobs", len(c.Options.CronJobs))
+	go func() {
+		// wait for signal and stop cron
+		<-c.Server.Stopper()
+		c.Cron.Stop()
+	}()
+	return nil
```

Comment:
[nit] I tend to prefer to have `return` with a newline unless it's the online line in a block